open-telemetry/opentelemetry-collector-contrib
# spanmetricsprocessor support for delta aggregation temporality

Tracked issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6484

### What 

Adds support for `spanmetricsprocessor` to generate metrics with delta aggregation temporality. This can be configured by the user in the yaml, as written in the README. The default remains cumulative. 

### How 

After each metrics generation, the maps that hold the cumulative data are reset if the temporality is set to delta. Additionally, the metrics themselves are marked with the correct aggregation temporality. This is only a few lines of code change overall.

### Testing 

I have added a few tests, and restructured one of the existing tests to cover wider use cases. The test cases to verify correct metrics are: 
- Verify cumulative metrics in a single consumption. (existing) 
- Verify delta metrics in a single consumption. 
- Verify cumulative metrics in two consumptions (new, should accumulate)
- Verify delta metrics in two consumptions (should not accumulate)
